x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search_url_state.ts

```diff
@@ -49,16 +49,13 @@ const reducer = (state: HostsState, action: Action): HostsState => {
     case 'setFilter':
       return { ...state, filters: [...action.payload] };
     case 'setQuery':
-      const { filters, query, panelFilters, ...payload } = action.payload;
-      const newFilters = !filters ? state.filters : filters;
-      const newControlPanelFilters = !panelFilters ? state.panelFilters : panelFilters;
-      const newQuery = !query ? state.query : query;
+      const payload = Object.fromEntries(
+        Object.entries(action.payload).filter(([_, v]) => v !== undefined)
```

Comment:
Q: Can we have 
```suggestion
        Object.entries(action.payload).filter(([_, v]) => !!v)
```
(so it will match the previous condition checking with `!value`)?
Wdyt?